SocialCareCaseViewerApi.Tests/V1/Gateways/DatabaseGateway.Tests.cs

```diff
@@ -2330,6 +2330,49 @@ public void UpdatePersonSetsCorrectValuesForPersonEntity()
             person.ReviewDate.Should().Be(request.ReviewDate);
         }
 
+        [Test]
+        public void UpdatePersonsAddressUpdatesTheMostRecentAddressToDisplay()
+        {
+            var first = DateTime.Today;
+            var second = DateTime.Today.AddDays(-1);
+            var last = DateTime.Today.AddDays(-2);
+
+            var person = DatabaseGatewayHelper.CreatePersonDatabaseEntity();
+            person.Addresses = null;
+            var firstAddress = DatabaseGatewayHelper.CreateAddressDatabaseEntity(person.Id, startDate: first, isDisplayAddress: "Y");
+            var secondAddress = DatabaseGatewayHelper.CreateAddressDatabaseEntity(person.Id, startDate: second, isDisplayAddress: "Y");
+            var lastAddress = DatabaseGatewayHelper.CreateAddressDatabaseEntity(person.Id, startDate: last, isDisplayAddress: "Y");
+
+            DatabaseContext.Persons.Add(person);
+            DatabaseContext.Addresses.Add(firstAddress);
+            DatabaseContext.Addresses.Add(secondAddress);
+            DatabaseContext.Addresses.Add(lastAddress);
+            DatabaseContext.SaveChanges();
+
+            var personBeforeUpdate = _classUnderTest.GetPersonDetailsById(person.Id);
```

Comment:
If the implementation of this method changes then this test will potentially be affected. Would it be safer to use the DatabaseContext directly when fetching records?